site/src/Map.jsx

```diff
@@ -1,135 +1,313 @@
-import './CustomControls.css';
-import ThemeSelector from './ThemeSelector';
-
-const MAP_STYLE = 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json';
-const DARK_MAP_STYLE = 'https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json'
-
-const PLACES_PMTILES_URL = 'pmtiles://https://data.source.coop/protomaps/overture/2024-04-16-beta.0/places.pmtiles';
-const PLACES_MAP_STYLE = {
-  'id': 'places',
-  'type': 'circle',
-  'source': 'places',
-  'source-layer': 'places',
-  'filter': [">=", ["get", "confidence"], 0.0],
-  'paint': {
-    'circle-color':
-      ['case',
-        ['==', ['get', '@category'], 'beauty_salon'], '#fb9a99',
-        ['==', ['get', '@category'], 'hotel'], '#33a02c',
-        ['==', ['get', '@category'], 'landmark_and_historical_building'], '#a6cee3',
-        ['==', ['get', '@category'], 'professional_services'], '#fdbf6f',
-        ['==', ['get', '@category'], 'shopping'], '#e31a1c',
-        ['==', ['get', '@category'], 'restaurant'], '#1f78b4',
-        ['==', ['get', '@category'], 'school'], '#ff7f00',
-        ['==', ['get', '@category'], 'accommodation'], '#b2df8a',
-        '#cab2d6'
-      ],
-    'circle-radius': [
-      'interpolate',
-      ['exponential', 2],
-      ['zoom'],
-      0, 1,
-      19, 8
-    ],
-    'circle-stroke-width': [
-      'interpolate',
-      ['exponential', 2],
-      ['zoom'],
-      12, 0,
-      14, 2
-    ],
-    'circle-stroke-color': 'black'
-  }
-};
+import {
+  Map as MapLibreMap,
+  NavigationControl,
+  Source,
+  AttributionControl,
+} from "react-map-gl/maplibre";
+import "maplibre-gl/dist/maplibre-gl.css";
+import * as pmtiles from "pmtiles";
+import maplibregl from "maplibre-gl";
+import { useState, useEffect, useCallback, useRef } from "react";
+import { Layer, GeolocateControl } from "react-map-gl/maplibre";
+import InspectorPanel from "./InspectorPanel";
+import PropTypes from "prop-types";
+import "./InspectorPanel.css";
+import "./CustomControls.css";
+import ThemeSelector from "./ThemeSelector";
 
+const PMTILES_URL =
+  "pmtiles://https://data.source.coop/protomaps/overture/2024-05-16-beta.0/";
 
 const INITIAL_VIEW_STATE = {
-  latitude: 51.0500,
+  latitude: 51.05,
   longitude: 3.7303,
   zoom: 16,
   bearing: 0,
-  pitch: 0
+  pitch: 0,
+};
+
+// this reference must remain constant to avoid re-renders
+const MAP_STYLE = {
+  version: 8,
+  glyphs: "https://demotiles.maplibre.org/font/{fontstack}/{range}.pbf",
+  sources: {},
+  layers: [],
+};
+
+const ThemeSource = ({ name, url }) => {
+  return <Source id={name} type="vector" url={`${url}${name}.pmtiles`} />;
+};
+
+ThemeSource.propTypes = {
+  name: PropTypes.string.isRequired,
+  url: PropTypes.string.isRequired,
+};
+
+const ThemeTypeLayer = ({
+  theme,
+  type,
+  color,
+  point,
+  line,
+  polygon,
+  extrusion,
+}) => {
+  return (
+    <>
+      {point ? (
+        <Layer
+          filter={["==", ["geometry-type"], "Point"]}
+          beforeId="divisions_division"
```

Comment:
Why do you need this `beforeId`? I would assume that rendering the Layers after the `divisions_division` layer would already accomplish this.

Reply:
The order of `Layer` in the JSX tree does not seem to correspond to their visual order in `react-map-gl`. See https://visgl.github.io/react-map-gl/docs/api-reference/layer This is surprising since JSON styles do follow the exact order. I suppose this is another case where toggling layers via `visibility` may be preferred over removing/re-mounting them. 

In the PR as is the layer Z-order is inconsistent, that `beforeId` is there to keep the label on top of everything.

Reply:
It looks like toggling `layout.visibility` is indeed the preferred solution: https://github.com/visgl/react-map-gl/issues/939

I will update the PR to use that method.

Reply:
Updated to use `visibility` toggles which preserves layer ordering, and removed `beforeId`.